pkg/executor/point_get.go

```diff
@@ -687,22 +699,29 @@ func (e *PointGetExecutor) get(ctx context.Context, key kv.Key) ([]byte, error)
 	lock := e.tblInfo.Lock
 	if lock != nil && (lock.Tp == ast.TableLockRead || lock.Tp == ast.TableLockReadOnly) {
 		if e.Ctx().GetSessionVars().EnablePointGetCache {
```

Comment:
I think we can put the check here

```suggestion
		if e.Ctx().GetSessionVars().EnablePointGetCache && e.commitTSOffset < 0 {
```